libs/sof/epipolar_curves.cpp

```
@@ -64,16 +64,18 @@ void EpipolarCurves::Candidates(const Vector2T& uv_l_base, std::vector<Vector2T>
   out.clear();
   const float v_f = uv_l_base.y() * inv_scale_;
   const float u_f = uv_l_base.x() * inv_scale_;
-  // Guard against negative, NaN, or oversized inputs. The `!(x >= 0.f)` form catches NaN too.
-  if (!(v_f >= 0.f) || !(u_f >= 0.f)) {
+
+  const size_t top_image_w = curves_[0].size();
+  const size_t top_image_h = curves_.size();
+  if (v_f < 0.f || v_f >= static_cast<float>(top_image_h) || u_f < 0.f || u_f >= static_cast<float>(top_image_w)) {
```

Clang (14.0.6)

[warning] floating point literal has suffix 'f', which is not uppercase

(readability-uppercase-literal-suffix)